zeroex/order_validator_test.go

```diff
@@ -251,6 +252,154 @@ func TestBatchValidateCoordinatorSoftCancels(t *testing.T) {
+	require.NoError(t, err)
+
+	ethClient, err := ethclient.Dial(constants.GanacheEndpoint)
+	require.NoError(t, err)
+
+	maxContentLengthAboveSingleOrder := singleOrderPayloadSize + 1
+
+	orderValidator, err := NewOrderValidator(ethClient, constants.TestNetworkID, maxContentLengthAboveSingleOrder)
+	require.NoError(t, err)
+
+	expectedIsBelow := true
+	isBelow := orderValidator.isBelowMaxContentLength([]*SignedOrder{signedOrder})
+	require.NoError(t, err)
+	assert.Equal(t, expectedIsBelow, isBelow)
+}
+
+func TestFindChunkSizeReturnLow(t *testing.T) {
+	signedOrder, err := SignTestOrder(&testSignedOrder.Order)
+	require.NoError(t, err)
+
+	ethClient, err := ethclient.Dial(constants.GanacheEndpoint)
+	require.NoError(t, err)
+
+	maxContentLengthFitsThreeOrders := singleOrderPayloadSize * 3
+
+	orderValidator, err := NewOrderValidator(ethClient, constants.TestNetworkID, maxContentLengthFitsThreeOrders)
+	require.NoError(t, err)
+
+	signedOrders := []*SignedOrder{}
+	for i := 0; i < 5; i++ {
+		signedOrders = append(signedOrders, signedOrder)
+	}
+
+	expectedChunkSize := int64(3)
+	chunkSize := orderValidator.findChunkSize(signedOrders)
+	require.NoError(t, err)
+	assert.Equal(t, expectedChunkSize, chunkSize)
+}
+
+func TestFindChunkSizeReturnHigh(t *testing.T) {
+	signedOrder, err := SignTestOrder(&testSignedOrder.Order)
+	require.NoError(t, err)
+
+	ethClient, err := ethclient.Dial(constants.GanacheEndpoint)
+	require.NoError(t, err)
+
+	maxContentLengthFitsFourOrders := singleOrderPayloadSize * 4
+
+	orderValidator, err := NewOrderValidator(ethClient, constants.TestNetworkID, maxContentLengthFitsFourOrders)
+	require.NoError(t, err)
+
+	// When number of orders in below maxContentLength, the binary-search always sets hi to mid. Hi then
+	// contains the final result.
+	signedOrders := []*SignedOrder{}
+	for i := 0; i < 2; i++ {
+		signedOrders = append(signedOrders, signedOrder)
+	}
+
+	expectedChunkSize := int64(2)
+	chunkSize := orderValidator.findChunkSize(signedOrders)
+	require.NoError(t, err)
+	assert.Equal(t, expectedChunkSize, chunkSize)
+}
+
+// TODO: Catch throw?
+func TestFindChunkSizeFails(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Expected findChunkSize() to panic if maxContentLength doesn't fit even a single order")
+		}
+	}()
+
+	signedOrder, err := SignTestOrder(&testSignedOrder.Order)
+	require.NoError(t, err)
+
+	ethClient, err := ethclient.Dial(constants.GanacheEndpoint)
+	require.NoError(t, err)
+
+	maxContentLengthFitsZeroOrders := singleOrderPayloadSize - 100
+
+	orderValidator, err := NewOrderValidator(ethClient, constants.TestNetworkID, maxContentLengthFitsZeroOrders)
+	require.NoError(t, err)
+
+	signedOrders := []*SignedOrder{}
+	for i := 0; i < 5; i++ {
+		signedOrders = append(signedOrders, signedOrder)
+	}
+
+	expectedChunkSize := int64(0)
+	chunkSize := orderValidator.findChunkSize(signedOrders)
+	assert.Equal(t, expectedChunkSize, chunkSize)
```

Comment:
This assertion will never actually run if `findChunkSize` panics. You can just remove it.